### PR TITLE
intermediate cli

### PR DESCRIPTION
**Pull request by tozanski**

### CODE REVIEW

File: julia/src/intermediate_cli.jl [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    mild_detection_prob = mild_detection_prob,
+        
+    constant_kernel_param = constant_kernel_param,
+    household_kernel_param = household_kernel_param,
+    hospital_kernel_param = hospital_kernel_param,
+        
+    backward_tracking_prob = tracking_prob,
+    backward_detection_delay = tracking_backward_delay,
+        
+    forward_tracking_prob = tracking_prob,
```
Comment (eMaerthin) [inline julia/src/intermediate_cli.jl:57]: is this intentionally `tracking_prob` not `forward_tracking_prob`?

File: julia/src/intermediate_cli.jl [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    rng,
+    population = individuals_df, 
+        
+    mild_detection_prob = mild_detection_prob,
+        
+    constant_kernel_param = constant_kernel_param,
+    household_kernel_param = household_kernel_param,
+    hospital_kernel_param = hospital_kernel_param,
+        
+    backward_tracking_prob = tracking_prob,
```
Comment (eMaerthin) [inline julia/src/intermediate_cli.jl:54]: is this intentionally `tracking_prob` not `backward_tracking_prob`?

File: julia/src/intermediate_cli.jl [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    backward_detection_delay = tracking_backward_delay,
+        
+    forward_tracking_prob = tracking_prob,
+    forward_detection_delay = tracking_forward_delay,
+        
+    testing_time = tracking_testing_delay,
```
Comment (eMaerthin) [inline julia/src/intermediate_cli.jl:60]: why `tracking_testing_delay` not `testing_time`?

File: julia/src/intermediate_cli.jl [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+  fill!(cb.detection_times, missing)
+  fill!(cb.detection_types, 0)
+  fill!(cb.tracking_sources, 0)
+  cb.num_infected_remain = max_num_infected
+end
+
+isdetection(ek::Simulation.EventKind) = ek == Simulation.DetectedOutsideQuarantineEvent || ek == Simulation.DetectedFromTrackingEvent || ek == Simulation.DetectedFromQuarantineEvent
+istransmission(ek::Simulation.EventKind) = ek == Simulation.TransmissionEvent || ek == Simulation.OutsideInfectionEvent
+isquarantine(ek::Simulation.EventKind) = ek == Simulation.QuarantinedEvent || ek == Simulation.QuarantineEndEvent
+ishospitalized(ek::Simulation.EventKind) = ek == Simulation.GoHospitalEvent
```
Comment (eMaerthin) [inline julia/src/intermediate_cli.jl:103]: not used?

File: julia/src/intermediate_cli.jl [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+isdeath(ek::Simulation.EventKind) = ek == Simulation.DeathEvent
```
Comment (eMaerthin) [inline julia/src/intermediate_cli.jl:104]: not used

File: julia/src/intermediate_cli.jl [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+function reset!(cb::DetectionCallback, max_num_infected::Integer=10^8)
+  fill!(cb.detection_times, missing)
+  fill!(cb.detection_types, 0)
+  fill!(cb.tracking_sources, 0)
+  cb.num_infected_remain = max_num_infected
+end
+
+isdetection(ek::Simulation.EventKind) = ek == Simulation.DetectedOutsideQuarantineEvent || ek == Simulation.DetectedFromTrackingEvent || ek == Simulation.DetectedFromQuarantineEvent
+istransmission(ek::Simulation.EventKind) = ek == Simulation.TransmissionEvent || ek == Simulation.OutsideInfectionEvent
+isquarantine(ek::Simulation.EventKind) = ek == Simulation.QuarantinedEvent || ek == Simulation.QuarantineEndEvent
```
Comment (eMaerthin) [inline julia/src/intermediate_cli.jl:102]: not used

File: julia/src/intermediate_cli.jl [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+    cb.tracking_sources[subject] = Simulation.source(event)
+  end
+  return cb.num_infected_remain>0
+end
+
+const comma = ','    
+const float_with_offset(offset, time) = float(offset + time)
+
+
+function progression_csv_output(path::AbstractString, state::Simulation.SimState, params::Simulation.SimParams)
```
Comment (eMaerthin) [inline julia/src/intermediate_cli.jl:131]: not used

File: julia/src/intermediate_cli.jl [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+                  comma, float_with_offset(t, p.severe_symptoms_time), comma, float_with_offset(t, p.recovery_time), 
+                  comma, float_with_offset(t, p.death_time))
+    end
+  finally
+    close(io)  
+  end
+  nothing
+end
+
+function callback_csv_output(path::AbstractString, cb::DetectionCallback)
```
Comment (eMaerthin) [inline julia/src/intermediate_cli.jl:155]: not used

File: julia/src/intermediate_cli.jl [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+  param_save_path = output_prefix*"run_params.jld2"
+  
+  @info "saving parameters" num_individuals param_save_path
+  jldopen(param_save_path, "w") do dict
+    Simulation.saveparams(dict, params)
+  end
+
+  #
+  #states = Vector{Simulation.SimState}(undef, nthreads())
+  #@threads for i in 1:nthreads()
```
Comment (eMaerthin) [inline julia/src/intermediate_cli.jl:216]: this probably can be safely removed

File: julia/src/intermediate_cli.jl (hunk 2 above)
Comment (tozanski) [reply julia/src/intermediate_cli.jl:54]: right now it is same for both forward and backward

**Pull request merged**